### PR TITLE
Build optimization: conditionally skip integraiton tests

### PR DESCRIPTION
**Pull request by kpavlov**
## Issue
Integration tests build fails for PRs opened from forks due to gh-actions secrets limitation, cancelling unit test. Fork builds are always red.

The changes in this pull request aim to enhance the build and test workflows by running integration tests only when secrets are present, and allowing builds to run on the main branch when it is not a pull request.

Related to #1909

## Change

This pull request includes changes to the `.github/workflows/main.yaml` file to enhance the build and test workflows.
Run integration tests only:
- on the main branch when it is not a pull request
- for pull requests from the same repository
- when triggered manually

Enhancements to workflow triggers and conditions:

* [`.github/workflows/main.yaml`](diffhunk://#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3R26): Added `workflow_dispatch` to allow manual triggering of the workflow.
* [`.github/workflows/main.yaml`](diffhunk://#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3L59-R72): Refined the conditions for running integration tests to ensure they run only on the main branch, for pull requests from the same repository, or when triggered manually.

Performance improvements:

* [`.github/workflows/main.yaml`](diffhunk://#diff-71cabc4177e41ea8f15a89eb65398fa8187739f68a2762706d84885cd8b2ddc3L59-R72): Updated the Maven command to use 8 threads instead of 4 for faster builds.

## General checklist

- [x] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new maven module

- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -56,13 +57,19 @@ jobs:
...
 
       - name: Integration test with JDK ${{ matrix.java_version }}
-        if: ${{ github.repository == 'langchain4j/langchain4j' }}
+        ## allow builds to run only:
+        ## - on the main branch when it is not a pull request,
+        ## - for pull requests from the same repository
+        ## - when triggered manually
+        if: (github.event_name == 'pull_request' && github.head_repo.full_name == github.repository) || (github.event_name == 'push' && github.ref == 'refs/heads/main') || github.event_name == 'workflow_dispatch'
         run: |
+          if {{ env.OPENAI_API_KEY
```
Comment (langchain4j) [inline .github/workflows/main.yaml]: Is this line correct and required?

**Review by ghost — COMMENTED** [review]
@kpavlov thank you!

File: .github/workflows/main.yaml (hunk 1 above)
Comment (kpavlov) [reply .github/workflows/main.yaml]: nop, it's not needed

Comment (kpavlov) [reply .github/workflows/main.yaml]: @langchain4j , removed it

**Review by ghost — APPROVED** [review]
@kpavlov thank you!

**Pull request merged**